### ISSUE TITLE
Getting "Super expression must either be null or a function" After Building And Deploying

### ISSUE THREAD

**Issue by cngeru** [alexplumb/material-ui-phone-number#32]
I'm gettting `Super expression must either be null or a function`
Versions
"@material-ui/core": "^4.4.0",
"react": "^16.9.0",
 "material-ui-phone-number": "^2.2.2",

**Comment by cngeru** [comment]
`
import MuiPhoneNumber from 'material-ui-phone-number';
const AddPhoneNumber = (props) => { 
  const classes = useStyles();
  const [phoneNumber, setPhoneNumber] = useState(null);
  const [country, setCountry] = useState(null);
  const [loading, setLoading] = useState(false);
  const handleOnChange =(value,country)=>{
    setPhoneNumber(value)
    setCountry(country.name)
  }
  const handlePhoneNumber = async()=>{
    setLoading(true)
    try {
      const addPhoneNumber = functions.httpsCallable('addPhoneNumber');  
      await addPhoneNumber({ phoneNumber:phoneNumber,country:country }); 
      await auth.currentUser.getIdTokenResult(true)
      return window.location.reload()
    } catch (error) {
      console.log(error)
    }
  }

  return (
    <Paper className={classes.root} elevation={12}>
      <Typography variant="h6" gutterBottom style={{marginBottom:15,marginLeft:10}}>
        Please Add Your Phone Number Below
      </Typography>
      <MuiPhoneNumber
        required
        id="phoneNumber"
        label="Phone Number"
        autoComplete="tel"
        value={phoneNumber}
        margin="normal"
        variant="outlined"
        regions={'africa'}
        defaultCountry={'ke'} 
        onChange={(value,country)=>handleOnChange(value,country)}
      />
       {loading && <LinearProgress color="secondary" style={{width:'100%',margin:5}}/>}
      <div className={classes.bottom}>
        <Button
          disabled={!phoneNumber}
          variant="text" 
          color="primary" 
          size="medium"
          style={{color:'white'}}
          onClick={handlePhoneNumber}
          >
          Add phone Number
        </Button>
      </div>
    </Paper>
  );
};

export default withRouter(AddPhoneNumber);


`

**Comment by dfkoh** [comment]
I ran into the same issue as well, can provide a bit more detail. I ultimately worked around the issue by reverting to version 2.0.1 of the package, which seems to be fine.

The error only happens on the production build of our CRA app (development build is fine). It looks from the traceback like somehow an import is getting mangled in the minimization process, perhaps webpack is the culprit somehow?

I didn't have time to put together a minimal testcase but if you do a production CRA build on https://github.com/baltimorecorps/hippo-frontend at commit da220b80be64eb1999c7cfefc6e296398168b531, it should reliably reproduce the error.

**Comment by alexplumb** [comment]
@dfkoh Can you try changing `InputProps` to `inputProps` (no capitals) - I just want to eliminate that as a possible cause of this. You can see in the changelog (https://github.com/alexplumb/material-ui-phone-number/blob/master/CHANGELOG.md#210---2019-08-27) that this parameter had a breaking change in version 2.1.0, but it doesn't look like your code was updated.

**Comment by cngeru** [comment]
installed `lodash` it works fine apparently

**Comment by pablomartinezm** [comment]
I'm having the same issue. It works on development environment, but it returns `TypeError: Super expression must either be null or a function` after production build. We already have lodash in package.json, let's see if someon has the clue of how to solve this...

**Comment by alexplumb** [comment]
Just want to confirm - do you also have `classnames` installed? Can I see your package.json?

**Comment by pablomartinezm** [comment]
```js
{

  "dependencies": {
    "@material-ui/core": "4.5.0",
    "@material-ui/icons": "4.4.3",
    "axios": "^0.18.0",
    "chart.js": "^2.8.0",
    "classnames": "^2.2.6",
    "eslint-plugin-import": "^2.17.3",
    "eslint-plugin-react": "^7.11.0",
    "history": "^4.9.0",
    "lodash": "^4.17.11",
    "material-ui-phone-number": "^2.2.2",
    "mixpanel-browser": "^2.27.1",
    "moment": "^2.24.0",
    "pdfmake": "^0.1.60",
    "prop-types": "^15.7.2",
    "react": "^16.8.6",
    "react-beautiful-dnd": "^11.0.4",
    "react-chartjs-2": "^2.7.6",
    "react-dom": "^16.8.6",
    "react-dropzone": "^10.1.4",
    "react-freshchat": "^1.0.11",
    "react-geolocated": "^2.5.0",
    "react-intl": "^2.8.0",
    "react-joyride": "^2.1.0",
    "react-lottie": "^1.2.3",
    "react-preloading-screen": "^1.0.2",
    "react-redux": "^7.0.2",
    "react-router-dom": "^5.0.0",
    "react-scripts": "3.0.0",
    "react-stripe-elements": "^4.0.0",
    "react-swipe": "^6.0.4",
    "react-syntax-highlighter": "^11.0.2",
    "read-excel-file": "^3.0.0",
    "redux": "^4.0.1",
    "redux-thunk": "^2.3.0",
   
    "xlsx": "^0.14.3"
  },
  "scripts": {
    
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "babel-eslint": "^10.0.1",
    "eslint": "^5.16.0",
    "eslint-config-airbnb": "17.1.0",
    "eslint-config-airbnb-base": "^13.1.0",
    "eslint-config-prettier": "^4.1.0",
    "eslint-plugin-jsx-a11y": "^6.1.1",
    "eslint-plugin-prettier": "^3.0.1",
    "flow-bin": "^0.97.0",
    "flow-remove-types": "^1.2.3",
    "prettier": "^1.17.0"
  }
}
```

**Comment by alexplumb** [comment]
Can you try upgrading your version of lodash?

**Comment by pablomartinezm** [comment]
I upgraded it to `4.17.15` and still not works 😩

**Comment by josh-feldman** [comment]
Also getting this issue with lodash `4.17.15`, classnames `2.2.6`, and and no `InputProps`

**Comment by pablomartinezm** [comment]
I downgraded `material-ui-phone-number` to "2.0.1" and it's working now

**Comment by alexplumb** [comment]
Hey folks, sorry for the delay on this - was on vacation in Singapore for a couple weeks. I'll see what I can do today.

**Comment by alexplumb** [comment]
This should be resolved in `material-ui-phone-number@2.2.4`
